bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/handler/DaikinAcUnitHandler.java

```diff
@@ -44,49 +50,93 @@
  *
  * @author Tim Waterhouse - Initial Contribution
  * @author Paul Smedley <paul@smedley.id.au> - Modifications to support Airbase Controllers
- *
+ * @author Lukas Agethen - Added support for Energy Year reading, compressor frequency and powerful mode
  */
 @NonNullByDefault
 public class DaikinAcUnitHandler extends DaikinBaseHandler {
     private final Logger logger = LoggerFactory.getLogger(DaikinAcUnitHandler.class);
 
-    public DaikinAcUnitHandler(Thing thing, DaikinDynamicStateDescriptionProvider stateDescriptionProvider, @Nullable HttpClient httpClient) {
+    public DaikinAcUnitHandler(Thing thing, DaikinDynamicStateDescriptionProvider stateDescriptionProvider,
+            @Nullable HttpClient httpClient) {
         super(thing, stateDescriptionProvider, httpClient);
     }
 
     @Override
     protected void pollStatus() throws IOException {
-        ControlInfo controlInfo = webTargets.getControlInfo();
-        updateStatus(ThingStatus.ONLINE);
-        if (controlInfo != null) {
-            updateState(DaikinBindingConstants.CHANNEL_AC_POWER, controlInfo.power ? OnOffType.ON : OnOffType.OFF);
-            updateTemperatureChannel(DaikinBindingConstants.CHANNEL_AC_TEMP, controlInfo.temp);
-
-            updateState(DaikinBindingConstants.CHANNEL_AC_MODE, new StringType(controlInfo.mode.name()));
-            updateState(DaikinBindingConstants.CHANNEL_AC_FAN_SPEED, new StringType(controlInfo.fanSpeed.name()));
-            updateState(DaikinBindingConstants.CHANNEL_AC_FAN_DIR, new StringType(controlInfo.fanMovement.name()));
-
-            if (!controlInfo.power) {
-                updateState(DaikinBindingConstants.CHANNEL_AC_HOMEKITMODE, new StringType(HomekitMode.OFF.getValue()));
-            } else if (controlInfo.mode == Mode.COLD) {
-                updateState(DaikinBindingConstants.CHANNEL_AC_HOMEKITMODE, new StringType(HomekitMode.COOL.getValue()));
-            } else if (controlInfo.mode == Mode.HEAT) {
-                updateState(DaikinBindingConstants.CHANNEL_AC_HOMEKITMODE, new StringType(HomekitMode.HEAT.getValue()));
-            } else if (controlInfo.mode == Mode.AUTO) {
-                updateState(DaikinBindingConstants.CHANNEL_AC_HOMEKITMODE, new StringType(HomekitMode.AUTO.getValue()));
+        if (webTargets != null) {
+            ControlInfo controlInfo = webTargets.getControlInfo();
```

Comment:
Rather than forcing the indentation of every line I would rather you just exit early. And even then you should cache the variable locally to avoid the null checker warning.
```suggestion
        DaikinWebTargets webTargets = this.webTargets;
        if (webTargets == null) {
             return;
        }
        ControlInfo controlInfo = webTargets.getControlInfo();
```

Reply:
Good suggestion - implemented.